heracles/dices/shrinkage.py

```diff
@@ -103,76 +94,75 @@ def shrinkage_factor(cls1, target):
     return lambda_star
 
 
-def gaussian_covariance(Cls):
-    """
-    Computes Gaussian estimate of the target matrix.
-    input:
-        Cls: power spectra
-    returns:
-        T: target matrix
-    """
-    # Add bias to Cls
-    b = bias(Cls)
-    Cls = add_to_Cls(Cls, b)
-    # Separate Cls into Cls
-    _Cls = _fields2components(Cls)
-    # Compute Gaussian covariance
+def gaussian_covariance(cls):
+    b = bias(cls)
+    cls = add_to_Cls(cls, b)
     cov = {}
-    for key1, key2 in itertools.combinations_with_replacement(Cls, 2):
-        # covariance key
+    for key1, key2 in itertools.combinations_with_replacement(cls.keys(), 2):
         a1, b1, i1, j1 = key1
         a2, b2, i2, j2 = key2
         covkey = (a1, b1, a2, b2, i1, j1, i2, j2)
         # get reference results
-        cl1 = Cls[key1]
-        cl2 = Cls[key2]
+        cl1 = cls[key1]
+        cl2 = cls[key2]
         sa1, sb1 = cl1.spin
         sa2, sb2 = cl2.spin
-        # get components
-        _a1, idx1 = _split_key(a1, sa1, pos=0)
-        _b1, idx2 = _split_key(b1, sb1, pos=0)
-        _a2, idx3 = _split_key(a2, sa2, pos=0)
-        _b2, idx4 = _split_key(b2, sb2, pos=0)
+        # dof spins
+        dof_a1 = 1 if sa1 == 0 else 2
+        dof_b1 = 1 if sb1 == 0 else 2
+        dof_a2 = 1 if sa2 == 0 else 2
+        dof_b2 = 1 if sb2 == 0 else 2
```

Comment:
This is only to get the correct output shape, which we can now get from cl1.shape and cl2.shape directly.

```suggestion
```